server/src/main/java/org/opensearch/search/profile/query/QueryProfileBreakdown.java

```diff
@@ -42,15 +47,22 @@
  *
  * @opensearch.internal
  */
-public final class QueryProfileBreakdown extends ContextualProfileBreakdown<QueryTimingType> {
+public final class QueryProfileBreakdown extends ContextualProfileBreakdown {
 
-    /** Sole constructor. */
-    public QueryProfileBreakdown() {
-        super(QueryTimingType.class);
+    public QueryProfileBreakdown(Collection<Supplier<ProfileMetric>> metrics) {
+        super(metrics);
     }
 
     @Override
-    public AbstractProfileBreakdown<QueryTimingType> context(Object context) {
+    public ContextualProfileBreakdown context(Object context) {
         return this;
     }
+
+    public static Collection<Supplier<ProfileMetric>> getQueryTimers() {
```

Comment:
Not really pressing issue, but I would suggest to have an utility class, fe `ProfileMetrics` and move this method out there as `getDefaultQueryProfileMetrics()`